### PR TITLE
travis improvments

### PR DESCRIPTION
**Pull request by Primer42**
Turns on caching and container builds for travis.

Also, actually fail the tests when the ruby tests fail.

@roymarantz @byxorna @maddalab @defect @gtorre @sushruta

### CODE REVIEW

File: support/ruby/collins-client/.gitignore [hunk 1]
```diff
@@ -5,3 +5,5 @@ rdoc
 .yardoc
 doc
 coverage
+vendor
+.bundle
```
Comment (roymarantz) [inline support/ruby/collins-client/.gitignore]: why don't you want this in git?

**Comment by byxorna** [comment]
LGTM

File: support/ruby/collins-client/.gitignore (hunk 1 above)
Comment (Primer42) [inline support/ruby/collins-client/.gitignore]: We haven't in the past included them.  We could add them, but if we want to then I want to do it in a different PR
